### PR TITLE
fix:[UIUX-567] additional dimension of asset type in the asset trend graph

### PR DESCRIPTION
**Pull request by go-jay**
# Description

At present, we only display the general trend for assets without providing details on the specific asset types. Given that the necessary data is available in the backend, it would be valuable to incorporate insights into the distinct asset types within the same graph.

https://paladincloud.atlassian.net/browse/UIUX-567

Fixes # (issue)

- Developed a shared component for visualising asset trends, incorporating an asset type filter. 
- Common component was then implemented in both the asset graph on the dashboard and the asset summary section.
- Code streamlined, eliminating redundancy.
- Please review the attached video to observe and reference the behaviour 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also
list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] My commit message/PR follows the contribution guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# **Other information**:

https://github.com/PaladinCloud/CE/assets/152586069/6197e7b7-0bc4-428c-8781-f1dd23598802

### CODE REVIEW

**Review by plyubich — APPROVED** [review]

**Review by plyubich — CHANGES_REQUESTED** [review]
CE doesn't know anything about our plugins especially RedHat

File: webapp/src/app/shared/components/molecules/custom-card/custom-card.component.html [hunk 1]
```diff
@@ -1,6 +1,21 @@
 <mat-card class="panel">
     <mat-card-header>
-        {{ card.header }}
+        <span class="title">{{ card.header }}</span>
+        <div *ngIf="tabs">
+            <app-multi-tab-switcher [tabs]="tabs" [tabSelected]="tabSelected" (switchView)="switchTabView($event)"></app-multi-tab-switcher>
+        </div>
+        <app-table-filter-chip
+            *ngIf="filters?.list?.length !== 0 && appliedFiltersDict"
```
Comment (plyubich) [inline webapp/src/app/shared/components/molecules/custom-card/custom-card.component.html:8]: would be good to extract logic like this to a variable in ts file

File: webapp/src/app/shared/constants/global.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+// === Asset Groups ===
+
+export const REDHAT = 'redhat';
+export const CONTRAST = 'contrast';
```
Comment (plyubich) [inline webapp/src/app/shared/constants/global.ts:4]: Can't expose this in CE

**Review by plyubich — CHANGES_REQUESTED** [review]

File: webapp/src/app/shared/constants/global.ts (hunk 2 above)
Comment (go-jay) [reply webapp/src/app/shared/constants/global.ts]: Removed it

File: webapp/src/app/shared/components/molecules/custom-card/custom-card.component.html (hunk 1 above)
Comment (go-jay) [reply webapp/src/app/shared/components/molecules/custom-card/custom-card.component.html]: Done

**Review by plyubich — DISMISSED** [review]

**Pull request merged**